android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboardDownloaderActivity.java

```diff
@@ -158,429 +142,47 @@ public void onCreate(Bundle savedInstanceState) {
-      if (kbDownloadEventListeners != null) {
-        HashMap<String, String> keyboardInfo =
-          CloudDataJsonUtil.createKeyboardInfoMap(pkgID,langID,langName,kbID,kbName,kbVersion,kbIsCustom,font,oskFont);
-
-        KeyboardEventHandler.notifyListeners(kbDownloadEventListeners, eventType, keyboardInfo, result);
-      }
-    }
-
-    /**
-     * Notify listeners when a lexical model is installed
-     * @param eventType
-     * @param models
-     * @param result
-     */
-    protected void notifyLexicalModelInstallListeners(KeyboardEventHandler.EventType eventType,
-      List<Map<String, String>> models, int result) {
-      if (kbDownloadEventListeners != null) {
-        KeyboardEventHandler.notifyListeners(kbDownloadEventListeners, eventType, models, result);
-      }
-    }
-  }
-
-  /**
-   * Async task to download a Keyman keyboard from either Keyman cloud server or custom url
-   * @param context
-   * @param showProgressDialog
-   */
-  public static void download(final Context context, final boolean showProgressDialog) {
-    if(USE_DOWNLOAD_MANAGER)
-      downloadKeyboardUsingDownloadManager(context);
-    else
-      new DownloadTask(context, showProgressDialog).execute();
-  }
-
-  public static void download(final Context context, final boolean showProgressDialog,
-                              final boolean aDownloadOnlyLexicalModel) {
-    if(USE_DOWNLOAD_MANAGER)
-      downloadUsingDownloadManager(context,aDownloadOnlyLexicalModel);
-    else
-      new DownloadTask(context, showProgressDialog, aDownloadOnlyLexicalModel).execute();
-  }
-
-  /**
-   * Download in keyboards and lexical model using download manager.
-   * @param context the context
-   * @param aDownloadOnlyLexicalModel is lexical model download
+   * prepare cloud api params for lexical models download.
+   * @return the result
    */
-  private static void downloadUsingDownloadManager(final Context context,
-                              final boolean aDownloadOnlyLexicalModel)
-  {
-    if(aDownloadOnlyLexicalModel)
-    {
-      downloadLexicalModelUsingDownloadManager(context);
-    }
-    else
-    {
-      downloadKeyboardUsingDownloadManager(context);
-    }
+  private ArrayList<CloudApiTypes.CloudApiParam> prepareCloudApiParamsForLexicalModelDownload() {
+    ArrayList<CloudApiTypes.CloudApiParam> _params = new ArrayList<>();
+    _params.add(new CloudApiTypes.CloudApiParam(
+      CloudApiTypes.ApiTarget.LexicalModelPackage, url));
+    return _params;
   }
 
   /**
    * prepare and execute keyboard download using downloadmanager.
    * @param context the context
+   * @param aLangId the language id
+   * @param aKbId the keyman keyboard id
+   * @param aPreparedCloudApiParams the prepared api params
    */
-  private static void downloadKeyboardUsingDownloadManager(Context context)
+  public static void downloadKeyboardUsingDownloadManager(Context context,
```

Comment:
```suggestion
  public static void downloadKeyboard(Context context,
```

Now that you've cleaned up the deprecated download task, you can just simply name these `downloadKeyboard` and `downloadLexicalModel` (line 254)